### DISCUSSION TITLE
Websocket don't work

### DISCUSSION THREAD

**Discussion by Fruchtgummi** [99designs/gqlgen#1802] [Q&A]
### versions
 - `gqlgen version`?  0.13.0
 - `go version`? go1.16.3

 
```
cor := cors.New(cors.Options{
		AllowedOrigins:   []string{"*"},
		AllowCredentials: true,
	})

	s, err := NewGraphQLServer(config.Get().ServiceHostnameAdmin)
	if err != nil {
		log.Println(err)
	}

	router := chi.NewRouter()

	qlhandler := handler.NewDefaultServer(s.ToExecutableSchema())
	qlhandler.Use(extension.Introspection{})

	router.Handle("/healthcheck", auth.Healtcheck(qlhandler))
	router.Handle("/graphql", cor.Handler(auth.JwtAuth(qlhandler)))
	router.Handle("/playground", playground.Handler("ripQL", "/graphql"))

	var wait time.Duration
	srv := &http.Server{
		MaxHeaderBytes: 1 << 20,
		ReadTimeout:    180 * time.Second,
		WriteTimeout:   180 * time.Second,
		Addr:           "0.0.0.0:8080",
		Handler: router,
	}

	
	go func() {
		if err := srv.ListenAndServe(); err != nil {
			log.Println("Error: " + err.Error())
		}
	}()

	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	<-c

	ctx, cancel := context.WithTimeout(context.Background(), wait)
	defer cancel()

	srv.Shutdown(ctx)
	log.Println("Shutdown QL on mainloop")
	os.Exit(0)
```

Playground says 

```
{
  "error": "Could not connect to websocket endpoint ws://localhost:8080/graphql. Please check if the endpoint url is correct."
}
```

I don't know to use this case. What is the problem? All other requests, querys, mutations, run with no problems!

**Comment by Fruchtgummi** [comment]
The Bearer-Token is not sending by header. I need the JWT auth on my header... Insomnia and Postman don't works with websocket... Does anyone know any other tool to test graphQL+Bearer with websockets?

![image](https://user-images.githubusercontent.com/2059888/123555808-50706780-d788-11eb-99ff-a46ce397a907.png)

**Comment by ddouglas** [comment]
Check out this example to get Websockets working

https://github.com/99designs/gqlgen/blob/master/example/chat/server/server.go#L32-L43

**Comment by Fruchtgummi** [comment]
Yes, it' work, also by me too... But nothing with authenticate... The Playground send not header with Bearer-token..., this ist the problem. The Postman have websocket in beta, but send a header with Websocket:

![image](https://user-images.githubusercontent.com/2059888/123686748-6f3a3100-d850-11eb-8aaf-3d355e4b803f.png)

Gqlgen run with websocket, my JWT-Auth is too mutch for playground.

**Comment by RobertoOrtis** [comment]
That's because you need to set your headers to `"Sec-WebSocket-Protocol"` to `"graphql-ws"`

I am using gofiber:
```
c.Set("Sec-WebSocket-Protocol", "graphql-ws")
```
```
app.All("/query", func(c *fiber.Ctx) error {
    c.Set("Sec-WebSocket-Protocol", "graphql-ws")

    gqlHandler(c.Context())
    return nil
})
```

By the way, you don't need to do that when using Safari (it will work without it). However, in Chrome and Edge, won't work. Chrome won't report the correct error in the Console, and Edge does report it correctly.
